File: Data Structures & Algorithms/single-number-iii/submission-1.py
class Solution:
    def singleNumber(self, nums: List[int]) -> List[int]:
        n, res = len(nums), []

        for i in range(n):
            check = True
            for j in range(n):
                if i != j and nums[i] == nums[j]:
                    check = False
                    break
            if check:
                res.append(nums[i])
                if len(res) == 2:
                    break
        return res

File: Data Structures & Algorithms/top-k-elements-in-list/submission-1.cpp
class Solution {
public:
    vector<int> topKFrequent(vector<int>& nums, int k) {
        unordered_map<int, int> freq;
        for(int num : nums) {
            freq[num]++;
        }

        priority_queue<pair<int, int>> heap;
        for(auto& entry : freq) {
            heap.push({entry.second, entry.first});
        }

        vector<int> res;
        for(int i = 0; i < k && !heap.empty(); i++) {
            res.push_back(heap.top().second);
            heap.pop();
        }
        return res;
    }
};
